src/main/java/seedu/address/model/ingredient/Ingredient.java

```diff
@@ -33,9 +35,10 @@ public int getQuantity() {
     /**
      * Returns true if both ingredients have same name.
      */
-    public boolean isSameIngredient(Ingredient otherIngredient) {
-        if (otherIngredient.equals(this)) {
-            return true;
+    @Override
+    public boolean isSameItem(Object other) {
+        if (other instanceof Ingredient) {
+            return ingredientName.equals(((Ingredient) other).getIngredientName());
```

Comment:
Can just use ((Ingredient) other).ingredientName here
The getter method is unnecessary as you are in the same class